include/dd/Edge.hpp

```diff
@@ -1,14 +1,50 @@
 #pragma once
 
 #include "dd/Complex.hpp"
-#include "dd/ComplexValue.hpp"
 #include "dd/DDDefinitions.hpp"
 
-#include <array>
 #include <cstddef>
+#include <functional>
+#include <type_traits>
+#include <unordered_set>
 #include <utility>
 
 namespace dd {
+
+///-----------------------------------------------------------------------------
+///                          \n Forward declarations \n
+///-----------------------------------------------------------------------------
+struct vNode;
+struct mNode;
+struct dNode;
+
+///-----------------------------------------------------------------------------
+///                        \n Type traits and typedefs \n
+///-----------------------------------------------------------------------------
+template <typename T>
+using isVector = std::enable_if_t<std::is_same_v<T, vNode>, bool>;
+template <typename T>
+using isMatrix = std::enable_if_t<std::is_same_v<T, mNode>, bool>;
+template <typename T>
+using isDensityMatrix = std::enable_if_t<std::is_same_v<T, dNode>, bool>;
+template <typename T>
+using isMatrixVariant =
+    std::enable_if_t<std::is_same_v<T, mNode> || std::is_same_v<T, dNode>,
+                     bool>;
+
+using AmplitudeFunc =
+    std::function<void(const std::size_t, const std::complex<fp>&)>;
+using ProbabilityFunc = std::function<void(const std::size_t, const fp&)>;
+using MatrixEntryFunc = std::function<void(const std::size_t, const std::size_t,
+                                           const std::complex<fp>&)>;
+
+/**
+ * @brief A DD node with an edge weight
```

Comment:
Maybe describe as follows?

> A weighted edge pointing to a DD node.

Reply:
Good suggestion! Done in https://github.com/cda-tum/mqt-core/pull/434/commits/a2964ae7e6cecd95314d3fa9c2e0402bbf68c3ec